src/Fetch/Server.php

```diff
@@ -135,6 +142,33 @@ public function __construct($serverPath, $port = 143, $service = 'imap')
         $this->service = $service;
     }
 
+    public function __destruct()
+    {
+        $this->setImapErrors();
+    }
+
+    /**
+     * Returns all of the IMAP errors that have occurred
+     *
+     * @return array
+     */
+    public function getImapErrors()
+    {
+        if(empty($this->imapErrors)) {
+            $this->setImapErrors();
+        }
+
+        return $this->imapErrors;
+    }
+
+    /**
+     * Sets all the IMAP errors that have occurred
+     */
+    public function setImapErrors()
+    {
+        $this->imapErrors = imap_errors();
+    }
+
     /**
```

Comment:
I don't understand the point of this. You're running the "setImapErrors" in the destructor, which is kind of useless- the "this->imapErrors" property is not going to exist after the destructor is called.